vllm/core/scheduler.py

```diff
@@ -259,6 +259,49 @@ def create_empty(cls) -> "SchedulerPrefillOutputs":
         )
 
 
+class SequenceGroupMetadataCache:
```

Comment:
1. I feel we could make this cache a common class, and use it like metadata=PyObjectCache(SequenceGroupMetadata).
2. Please add docstring to the class and methods.

Reply:
Good idea! Introduced a single python object caching class and reused it in both cases.